truffle/src/main/ruby/core/rubinius/common/array.rb

```diff
@@ -612,6 +612,29 @@ def flatten!(level=-1)
     nil
   end
 
+
+  def inspect
+    return "[]".force_encoding(Encoding::US_ASCII) if @total == 0
+    comma = ", "
+    result = "["
+
+    return "[...]" if Thread.detect_recursion self do
+      each_with_index do |element, index|
+        temp = element.inspect
+        result.force_encoding(temp.encoding) if index == 0
+        result << temp << comma
+      end
+    end
+
+    Rubinius::Type.infect(result, self)
+    #result.shorten!(2) # MODIFIED We don't implement string shorten.
```

Comment:
We'd rather you left inspect as exactly in the Rubinius code, and then implement an empty `shorten!` as a shim - like this https://github.com/jruby/jruby/blob/master/truffle/src/main/ruby/core/shims.rb#L59-L60.

It's possible that this will cause problems later when we shimmed methods that silently do nothing, but at least they're easy to search for and understand. It's worked well for us so far.